File: mk_item_list/mk_item_list.js
frappe.ui.form.on('MK Item List', {
    onload: function(frm) {
        // set default filter values
        frappe.query_report.set_filter_value('disabled', 0);
    },
    refresh: function(frm) {
        // refresh the report after changing filters
        frappe.query_report.refresh();
    },
    item_group: function(frm) {
        // update the sub_group filter based on the selected item_group
        let item_group = frappe.query_report.get_filter_value('item_group');
        if (item_group) {
            frappe.db.get_list('Item Group', {
                fields: ['name'],
                filters: {
                    parent_item_group: item_group
                }
            }).then(data => {
                let sub_groups = data.map(d => d.name);
                frappe.query_report.set_filter_options('sub_group', sub_groups);
                frappe.query_report.set_filter_value('sub_group', null);
            });
        } else {
            frappe.query_report.set_filter_options('sub_group', []);
            frappe.query_report.set_filter_value('sub_group', null);
        }
    }
});

frappe.query_reports['Item Group and Item Name Report'] = {
    filters: [
        {
            fieldname: 'disabled',
            label: __('Disabled'),
            fieldtype: 'Check'
        },
        {
            fieldname: 'item_group',
            label: __('Item Group'),
            fieldtype: 'Link',
            options: 'Item Group'
        },
        {
            fieldname: 'sub_group',
            label: __('Sub Group'),
            fieldtype: 'Link',
            options: 'Item Group',
            get_query: function() {
                let item_group = frappe.query_report.get_filter_value('item_group');
                return {
                    filters: {
                        parent_item_group: item_group
                    }
                };
            }
        }
    ]
};

File: mk_item_list/mk_item_list.py
import frappe

def execute(filters=None):
    return ItemList(filters).run()
class ItemList(object):
    def __init__(self, filters=None):
        self.filters = frappe._dict(filters or {})
        self.columns = []
        self.data = []
        self.depth_map={}

    def run(self):
        self.get_columns()
        self.get_data()
        return self.columns, self.data
    def get_columns(self):
        self.columns = [
            {
            "fieldname": "item_group",
            "label": "Item Group",
            "fieldtype": "Link",
            "options": "Item Group",
            "width": 300
            },
            {
            "fieldname": "item_name",
            "label": "Item Name",
            "align": "left",
            "fieldtype": "Link",
            "options": "Item",
            "width": 300
            } ]
    def get_data(self):
        qry = """
                (
                select ig.name as item_group, i.name as item_name, ig.parent_item_group from `tabItem Group` 
                ig LEFT JOIN `tabItem` i ON ig.name=i.item_group order by lft                
                )
             """
        self.group_entries = frappe.db.sql(qry, as_dict=1)           
        for d in self.group_entries:
            if d.parent_item_group:
                self.depth_map.setdefault(d.item_group, self.depth_map.get(d.parent_item_group) + 1)
            else:
                self.depth_map.setdefault(d.item_group, 0)
        for d in self.group_entries:
            self.data.append({
                "item_group": d.item_group,
                "indent": self.depth_map.get(d.item_group,0),
                "item_name": d.item_name})
